client/gutenberg/extensions/subscriptions/edit.jsx

```diff
@@ -0,0 +1,166 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { Component, Fragment } from '@wordpress/element';
+import {
+	TextControl,
+	Button,
+	ToggleControl,
+	Disabled,
+	withFallbackStyles,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { __ } from 'gutenberg/extensions/presets/jetpack/utils/i18n';
+import { sprintf } from '@wordpress/i18n/build/index';
+import apiFetch from '@wordpress/api-fetch';
+import { compose } from '@wordpress/compose';
+import {
+	ContrastChecker,
+	InspectorControls,
+	withColors,
+	PanelColorSettings,
+} from '@wordpress/editor';
+
+const { getComputedStyle } = window;
+
+const applyFallbackStyles = withFallbackStyles( ( node, ownProps ) => {
+	const { textColor, backgroundColor } = ownProps;
+	const backgroundColorValue = backgroundColor && backgroundColor.color;
+	console.log( backgroundColorValue );
+	const textColorValue = textColor && textColor.color;
+	console.log( textColorValue );
+	// //avoid the use of querySelector if textColor color is known and verify if node is available.
+	const textNode =
+		! textColorValue && node ? node.querySelector( '[contenteditable="true"]' ) : null;
+	// return {
+	// 	fallbackBackgroundColor: backgroundColorValue || ! node ? undefined : getComputedStyle( node ).backgroundColor,
+	// 	fallbackTextColor: textColorValue || ! textNode ? undefined : getComputedStyle( textNode ).color,
+	// };
+} );
+
+console.log( applyFallbackStyles.textColor );
+
+class SubscriptionEdit extends Component {
+	render() {
+		const {
+			attributes,
+			className,
+			isSelected,
+			setAttributes,
+			backgroundColor,
+			textColor,
+			setBackgroundColor,
+			setTextColor,
+			fallbackBackgroundColor,
+			fallbackTextColor,
+		} = this.props;
+		const {
+			subscribe_placeholder,
+			show_subscribers_total,
+			subscriber_count_string,
+			button_background_color,
+			button_text_color,
+			button_large,
+		} = attributes;
+
+		const buttonStyle = {
+			'background-color': button_background_color,
+			color: button_text_color,
+			border: 'none',
+		};
+		// Get the subscriber count so it is available right away if the user toggles the setting
+		this.get_subscriber_count();
+
+		if ( isSelected ) {
+			return (
+				<Fragment>
+					<div className={ className } role="form">
+						{ isSelected && <p role="heading">{ subscriber_count_string }</p> }
+						<Disabled>
+							<TextControl placeholder={ subscribe_placeholder } required onChange={ () => {} } />
+						</Disabled>
+						<Button isDefault isLarge={ button_large } onClick={ null } style={ buttonStyle }>
+							{ __( 'Subscribe' ) }
+						</Button>
+					</div>
+					<InspectorControls>
+						<ToggleControl // Move this back to the block
+							label={ __( 'Show total subscribers' ) }
+							checked={ show_subscribers_total }
+							onChange={ () => {
```

Comment:
It makes sense to move any of those callbacks to separate methods, so we don't declare them on each re-render.